core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -285,6 +310,19 @@ public BytesToBytesMapIterator iterator() {
   }
 
   /**
+   * Returns an iterator for iterating over the entries of this map and sets if it is destructive.
+   *
+   * For efficiency, all calls to `next()` will return the same {@link Location} object.
+   *
+   * If any other lookups or operations are performed on this map while iterating over it, including
+   * `lookup()`, the behavior of the returned iterator is undefined.
+   */
+  public BytesToBytesMapIterator iterator(boolean destructive) {
```

Comment:
I would actually split this into two methods, one called `destructiveIterator` and another called just `iterator()`, like what we had before.  For the destructive version, I'd add a comment explaining that it's illegal to call _any_ other method on the map after the `destructiveIterator` method has been called.